spec/operators/catch-spec.ts

```diff
@@ -278,4 +279,84 @@ describe('Observable.prototype.catch', () => {
       done();
     });
   });
+
+  context('fromPromise', () => {
+    type SetTimeout = (callback: (...args: any[]) => void, ms: number, ...args: any[]) => NodeJS.Timer;
+
+    let trueSetTimeout: SetTimeout;
+    let sandbox: sinon.SinonSandbox;
+    let timers: sinon.SinonFakeTimers;
+
+    let source: Rx.Observable<any>;
+    beforeEach(() => {
```

Comment:
Hey @benlesh, thanks for taking a look at this PR. I'll update the tests to be more explicit per test and get back to you. I used sinon.sandbox as I found other examples in the current test suite: https://github.com/ReactiveX/rxjs/search?utf8=%E2%9C%93&q=sinon.sandbox&type=